### ISSUE TITLE
Doubleclick in Bookshelf opens two bibles

### ISSUE THREAD

**Issue by ilkant** [bibletime/bibletime#235]
Doubleclick is very common operation (Windows). And BibleTime 3.0 opens a work with one click. When you click twice, BibleTime opens two works to desktop. Many times it's pity. But sometimes it can be useful to open same work in two or more instances to desktop.

So the open click might have a short delay after it it takes next click to perform. The delay can be for example 1 second.

**Comment by jaakristioja** [comment]
I can not reproduce this with BibleTime 3.0 RC 1 on Gentoo Linux and KDE. Single- and double-clicking works well, with only double-click opening a work. Did you experience the issue with the BibleTime 3.0 RC 1 binary for Windows?

**Comment by gholmlund** [comment]
How this works is operating system dependent. Here is a quote from Qt about "activating" from a QTreeView.

[signal] void QAbstractItemView::activated(const QModelIndex &index)
This signal is emitted when the item specified by index is activated by the user. How to activate items depends on the platform; e.g., by single- or double-clicking the item, or by pressing the Return or Enter key when the item is current.

I did reproduce this by a very slow double click, which was really 2 single clicks. I don't believe there is anything we can change about this. I believe it is working the same as it has in previous versions.

**Comment by gholmlund** [comment]
Also, KDE and probably others allow you to choose single or double click to open files. It is in the system settings.

**Comment by jaakristioja** [comment]
@ilkant Is this also from the 3.0 RC1 snap package?

It might very well be that the snap package does not interact well with system settings from different desktop environments.

**Comment by ilkant** [comment]
Yes. I have heard that avoid snap. It installs but cant remove softrase.
Maybe that is true you said: "It might very well be that the snap package
does not interact well with system settings from different desktop
environments."

I have some amount of Kubuntu and Linux experience, so I can try to fix the
problem. But I think that it's important to have install package valid to
other BT people.

su 17. toukok. 2020 klo 22.54 Jaak Ristioja (notifications@github.com)
kirjoitti:

**Comment by ilkant** [comment]
I have installed BT 2.11.1 some years ago (about 1 or 2) from Kubuntu
repository. And now I have installed BT 3.0.0 RC1 twice. First BT 3.0.0
beta2 and then removed it with snap and then installed BT 3.0.0 RC1. I got
advices to do so from IRC #bibletime channel. Now I can start both 2.11.1
and 3.0.0 RC1 and they work in same time.

su 17. toukok. 2020 klo 23.06 Ilpo Kantonen (ilkantjabber@gmail.com)
kirjoitti:

**Comment by gholmlund** [comment]
I do reproduce this in the snap version of BT. It happens on kubuntu 18.04 and 20.04 and fedora 32 when using the snap package. Changing the OS single vs double click setting does not change the behavior. 
The snap is built using "core18" which is a ubuntu 18.04 build and runtime environment. I then built BT on 18.04 so it has the same qt and other runtime libraries as the snap. I could not reproduce the problem this way.
So, it appears to happen only with the snap package. The code that deals with single vs double click is in Qt, not in BT, so I doubt there is anything we can do about it.

**Comment by jaakristioja** [comment]
@gholmlund Perhaps something like the snapcraft [personal-files interface](https://snapcraft.io/docs/personal-files-interface) would help read such settings from the user's profile?

**Comment by jaakristioja** [comment]
@gholmlund, @ilkant: Can you still reproduce this with more recent versions of BibleTime, or can we close this?

**Comment by ilkant** [comment]
BT Version 3.0.3 opens two Bible windows. But I think that this is not very bad thing. You can close another window. No special need to build an unique open system.

**Comment by jaakristioja** [comment]
Ok. thanks! I'll close this then.
